snippets/standard/datetime/json/computing-with-jsondocument-valid.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Text.Json;
+
+public class Example
+{
+    private static double ComputeAverageTemperatures(string json)
+    {
+        var options = new JsonDocumentOptions
+        {
+            AllowTrailingCommas = true
+        };
+
+        using (JsonDocument document = JsonDocument.Parse(json, options))
+        {
+            int sumOfAllTemperatures = 0;
+            int count = 0;
+
+            foreach (JsonElement element in document.RootElement.EnumerateArray())
+            {
+                DateTimeOffset date = element.GetProperty("date").GetDateTimeOffset();
+
+                if (date.DayOfWeek == DayOfWeek.Monday)
+                {
+                    int temp = element.GetProperty("temp").GetInt32();
+                    sumOfAllTemperatures += temp;
+                    count++;
+                }
+            }
+
+            var averageTemp = (double)sumOfAllTemperatures / count;
+            return averageTemp;
+        }
+    }
+
+    public static void Main(string[] args)
+    {
+        string json =
+                @"[" +
+                    @"{" +
+                        @"""date"": ""2013-01-07T00:00:00Z""," +
+                        @"""temp"": 23," +
+                    @"}," +
+                    @"{" +
+                        @"""date"": ""2013-01-08T00:00:00Z""," +
+                        @"""temp"": 28," +
+                    @"}," +
+                    @"{" +
+                        @"""date"": ""2013-01-14T00:00:00Z""," +
+                        @"""temp"": 8," +
+                    @"}," +
+                @"]";
+
+        Console.WriteLine(ComputeAverageTemperatures(json));
```

Comment:
It isn't super clear/obvious from looking at the two samples, why one passes and one fails. However, I would imagine the text surrounding the sample will highlight that, so its probably fine.

I understand the localization concern, but does it still make sense to put a comment within/above the json string explicitly stating that the date time format is what's difference. At a quick glance, the two samples are almost identical otherwise.

@rpetrusha, what do you think?

Reply:
It's fine to add a comment in source code; I probably didn't make that clear, @ahsonkhan -- I didn't mean to imply that comments shouldn't be used. What I really meant is that comments should not be the only thing that illustrates something important in the example -- it's also important that the textual explanation of the example explains why one succeeds and the other fails.